### DISCUSSION TITLE
Internationalization module take precedence over basePath

### DISCUSSION THREAD

**Discussion by nigampatel** [vercel/next.js#22550] [Help]
I'm trying to figure out if this is possible in Next.js via config or by creating a custom server.

In my next.config.js I have `basePath` and `i18` enabled

`module.exports = {
    basePath: '/home',
    i18n: {
      locales: ['en-US', 'es-ES', 'fr-FR'],
      defaultLocale: 'en-US',
    },
}`

This gives me the following URL path `/home/en-US/`

Is there a way I can have the internationalization module take precedence over `basePath`, so basically I'm looking to have my URL to be `/en-US/home`?

**Comment by robertwbradford** [comment]
We are facing this same issue in adopting the App Router for our org. Trying to think of an API to propose that would allow the locale to exist before the basePath.
